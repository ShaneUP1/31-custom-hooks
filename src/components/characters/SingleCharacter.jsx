import React from 'react';
import PropTypes from 'prop-types';

const SingleCharacter = ({ name, image }) => (
  <figure>
    <img src={image} alt={name} />
    <figcaption>{name}</figcaption>
  </figure>
);

SingleCharacter.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default SingleCharacter;
