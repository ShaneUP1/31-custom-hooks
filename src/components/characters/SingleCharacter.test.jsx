/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SingleCharacter from './SingleCharacter';

describe('SingleCharacter component', () => {
  afterEach(() => cleanup());
  it('renders a single character', () => {
    const { asFragment } = render(<SingleCharacter name={'Ari'} image={'www.url.com'}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
