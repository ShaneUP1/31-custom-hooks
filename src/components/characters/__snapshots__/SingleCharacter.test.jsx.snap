// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleCharacter component renders a single character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Ari"
      src="www.url.com"
    />
    <figcaption>
      Ari
    </figcaption>
  </figure>
</DocumentFragment>
`;
